### ISSUE TITLE
Support deep merge

### ISSUE THREAD

**Issue by dankelleher** [julianduque/penv#3]
I would like to be able to do something like this:

package.json

```
{
  "dependencies": {
    "dep1": "1.0.0",
    "dep2": "1.0.0"
  }
}
```

environments.json
```
{
  "dev": {
    "dependencies": {
      "dep1": "1.2.0"
    }
  },
  "staging": {
    "dependencies": {
      "dep1": "1.1.0"
    }
  }
}
```

At the moment this is not possible, because the merge is shallow, so running `penv dev` with the above configuration would remove dep2.

**Comment by dankelleher** [comment]
PR here: https://github.com/julianduque/penv/pull/4

**Comment by julianduque** [comment]
@dankelleher landed in v0.2.0! Thanks!!! <3
